joern-cli/frontends/swiftsrc2cpg/src/test/scala/io/joern/swiftsrc2cpg/passes/ast/ClosureWithCompilerTests.scala

```diff
@@ -0,0 +1,258 @@
+package io.joern.swiftsrc2cpg.passes.ast
+
+import io.joern.swiftsrc2cpg.testfixtures.SwiftCompilerSrc2CpgSuite
+import io.shiftleft.codepropertygraph.generated.*
+import io.shiftleft.codepropertygraph.generated.nodes.*
+import io.shiftleft.semanticcpg.language.*
+
+class ClosureWithCompilerTests extends SwiftCompilerSrc2CpgSuite {
+
+  "ClosureWithCompilerTests" should {
+
+    "create type decls and bindings correctly (local closure declaration)" in {
+      val testCode =
+        """
+          |import Foundation
+          |
+          |func main() {
+          |  // closure call is allways "function_ref": "single_apply"
+          |
+          |	 let compare = { (s1: String, s2: String) -> Bool in
+          |    return s1 > s2
+          |	 }
+          |	 let compareResult = compare("1", "2")
+          |
+          |	 var customersInLine = ["Chris", "Alex", "Ewa", "Barry", "Daniella"]
+          |	 // auto-closure example 1:
+          |	 let customerProvider = { customersInLine.remove(at: 0) }
+          |	 let customerProviderResult = customerProvider()
+          |
+          |	 // auto-closure example 2:
+          |	 let greet = {
+          |    print("Hello, World!")
+          |	 }
+          |	 greet()
+          |
+          |	 let greetUser = { (name: String) in
+          |	   print("Hey there, \(name).")
+          |	 }
+          |	 greetUser("Alex")
+          |
+          |	 let findSquare = { (num: Int) -> (Int) in
+          |    let square = num * num
+          |	   return square
+          |	 }
+          |	 let findSquareResult = findSquare(5)
+          |}""".stripMargin
+
+      val cpg = codeWithSwiftSetup(testCode)
+
+      // compare ((Swift.String,Swift.String)->Swift.Bool)
+      val List(compareLocal) = cpg.local.nameExact("compare").l
+      compareLocal.typeFullName shouldBe "Swift.Function"
```

Comment:
We cannot just use a generic `Swift.Function` as type as this would conflate more functions into a single hierarchy then necessary, resulting in higher computational effort during e.g. call resolution. So we should use separate types for all the different function signatures. Same goes for all other places where `Swift.Function` is used.

Reply:
Signatures are now added to `Swift.Function` with surrounding `<...>`.